packages/smarthr-ui/src/components/Dialog/StepFormDialog/StepFormDialogContentInner.tsx

```diff
@@ -0,0 +1,183 @@
+
+export type BaseProps = PropsWithChildren<
+  DialogHeaderProps &
+    DialogBodyProps & {
+      /** アクションボタンのラベル */
+      submitLabel: ReactNode
+      /** アクションボタンのスタイル */
+      actionTheme?: 'primary' | 'secondary' | 'danger'
+      /**
+       * アクションボタンをクリックした時に発火するコールバック関数
+       * @param closeDialog ダイアログを閉じる関数
+       * @param currentStep onSubmitが発火した時のステップ
+       * @returns 次のステップに遷移する場合は次のステップ、遷移しない場合はundefined
+       */
+      onSubmit: (
+        closeDialog: () => void,
+        e: FormEvent<HTMLFormElement>,
+        currentStep: StepItem,
+      ) => StepItem | undefined
+      /** アクションボタンを無効にするかどうか */
+      actionDisabled?: boolean
+      /** 閉じるボタンを無効にするかどうか */
+      closeDisabled?: boolean
+      /** コンポーネント内の文言を変更するための関数を設定 */
+      decorators?: DecoratorsType<'closeButtonLabel' | 'nextButtonLabel' | 'backButtonLabel'>
+    }
+>
+
+export type StepFormDialogContentInnerProps = BaseProps & {
+  firstStep: StepItem
+  onClickClose: () => void
+  responseMessage?: ResponseMessageType
+  stepLength: number
+  onClickBack?: () => void
+}
+
+const CLOSE_BUTTON_LABEL = 'キャンセル'
+const NEXT_BUTTON_LABEL = '次へ'
+const BACK_BUTTON_LABEL = '戻る'
+
+export const StepFormDialogContentInner: FC<StepFormDialogContentInnerProps> = ({
+  children,
+  title,
+  titleId,
+  subtitle,
+  contentBgColor,
+  contentPadding,
+  submitLabel,
+  actionTheme = 'primary',
+  stepLength,
+  firstStep,
+  onSubmit,
+  onClickClose,
+  responseMessage,
+  actionDisabled = false,
+  closeDisabled,
+  decorators,
+  onClickBack,
+}) => {
+  const { currentStep, stepQueue, setCurrentStep } = useContext(StepFormDialogContext)
+  const activeStep = useMemo(() => currentStep?.stepNumber ?? 1, [currentStep])
+
+  const handleCloseAction = useCallback(() => {
+    onClickClose()
+    setTimeout(() => {
+      // HINT: ダイアログが閉じるtransitionが完了してから初期化をしている
+      stepQueue.current = []
+      setCurrentStep(firstStep)
+    }, 300)
+  }, [firstStep, stepQueue, setCurrentStep, onClickClose])
+
+  const handleSubmitAction = useCallback(
+    (e: FormEvent<HTMLFormElement>) => {
+      e.preventDefault()
+      // HINT: React Potals などで擬似的にformがネストしている場合など、stopPropagationを実行しないと
+      // 親formが意図せずsubmitされてしまう場合がある
+      e.stopPropagation()
+
+      stepQueue.current.push(currentStep)
+      const next = onSubmit(handleCloseAction, e, currentStep)
+      if (!next) {
+        return
+      }
+      setCurrentStep(next)
+    },
+    [currentStep, stepQueue, onSubmit, setCurrentStep, handleCloseAction],
+  )
+  const handleBackAction = useCallback(() => {
+    if (onClickBack) {
+      onClickBack()
+    }
+    const prev = stepQueue.current.pop() ?? firstStep
+    setCurrentStep(prev)
+  }, [firstStep, stepQueue, onClickBack, setCurrentStep])
+
+  const isRequestProcessing = responseMessage && responseMessage.status === 'processing'
+
+  const { wrapper, actionArea, buttonArea, message } = dialogContentInner()
```

Comment:
tvのclassName生成はuseMemoでラップしておいてもらえると 🙏 
```suggestion
  const classNames = useMemo(() => {
    const { wrapper, actionArea, buttonArea, message } = dialogContentInner()
    
    return {
      wrapper: wrapper(),
      actionArea: actionArea(),
      buttonArea: buttonArea(),
      message: message(),
    }
  }, [])
```

Reply:
利用箇所は `classNames.wrapper` のように変更してもらえると読みやすいと思います